parquet-thrift/src/main/java/org/apache/parquet/thrift/ThriftRecordConverter.java

```diff
@@ -445,6 +600,26 @@ public FieldEnumConverter(List<TProtocol> events, ThriftField field) {
       }
     }
 
+     @Override
+     protected ParquetProtocol decodeFromDictionary(Dictionary d, int i) {
+       Binary value = d.decodeToBinary(i);
+       final Integer id = enumLookup.get(value);
```

Comment:
I agree that these all follow the same pattern, and it looks like code duplication. However, they all extend a base class, and the only thing that IntConverter / LongConverter / etc. override is the part that has to be different for each.

IntConverter needs to call `dictionary.decodeToInt`, LongConverter needs to call `dictionary.decodeToInt`. IntConverter needs to return a protocol with an overidden `readI32` method, LongConverter needs to override `readI64`, etc.

The only thing I can see that could be de-duplicated is in:

```
static final class IntConverter extends PrimitiveConverterBase {

    IntConverter(List<TProtocol> events) {
      super(events);
    }

    @Override
    protected ParquetProtocol decodeFromDictionary(Dictionary d, int i) {
      final int value = d.decodeToInt(i);
      return new ParquetProtocol("readI32()") {
        @Override
        public int readI32() throws TException {
          return value;
        }
      };
    }

    @Override
    public void addInt(final int value) {
      events.add(new ParquetProtocol("readI32()") {
        @Override
        public int readI32() throws TException {
          return value;
        }
      });
    }
  }
```

we could instead do

```
static final class IntConverter extends PrimitiveConverterBase {

    IntConverter(List<TProtocol> events) {
      super(events);
    }
    
    private static ParquetProtocol event(final int value) {
      return new ParquetProtocol("readI32()") {
        @Override
        public int readI32() throws TException {
          return value;
        }
      };
    }    
    @Override
    protected ParquetProtocol decodeFromDictionary(Dictionary d, int i) {
      final int value = d.decodeToInt(i);
      return event(value);
    }

    @Override
    public void addInt(final int value) {
      events.add(event(value));
    }
  }
```

Is that what you mean?

Thanks,
Alex